$OpenBSD: patch-parser_c,v 1.1.1.1 2000/02/09 23:55:44 rohee Exp $

Was a core fixer in the previous version, should still be useful with
this one.

--- parser.c.orig	Thu Jan 13 12:53:06 2000
+++ parser.c	Mon Feb  7 21:34:16 2000
@@ -417,6 +417,9 @@ Bool InsertMisc(Node *element, Node *nod
 
 void ParseTag(Lexer *lexer, Node *node, uint mode)
 {
+    if (!node->tag)
+        return;
+
     if (node->tag->model & CM_EMPTY)
     {
         lexer->waswhite = no;
@@ -1542,7 +1545,7 @@ void ParseList(Lexer *lexer, Node *list,
             continue;
         }
 
-        if (node->tag != tag_li)
+        if ((node->tag != tag_li) && node->tag)
         {
             UngetToken(lexer);
 
